services/core-web/src/actionCreators/documentActionCreator.js

```diff
@@ -50,20 +53,29 @@ export const generateNoticeOfWorkApplicationDocument = (
       createRequestHeader()
     )
     .then((response) => {
-      const token = { token: response.data.token };
-      const docWindow = window.open(
-        `${ENVIRONMENT.apiUrl + API.RETRIEVE_CORE_DOCUMENT(token)}`,
-        "_blank"
-      );
-      docWindow.onbeforeunload = () => {
-        notification.success({
-          message,
-          duration: 10,
+      const params = { token: response.data.token, return_record: "true" };
+      return CustomAxios()
+        .get(`${ENVIRONMENT.apiUrl + API.DOCUMENT_GENERATION(params)}`, createRequestHeader())
+        .then((response) => {
+          const mineDocument = response.data;
+          notification.success({
+            message,
+            duration: 10,
+          });
+          dispatch(success(reducerTypes.GENERATE_NOTICE_OF_WORK_APPLICATION_DOCUMENT));
+          if (downloadGeneratedDocument) {
+            notification.info({
```

Comment:
Not sure if we want this notification message or not

Reply:
Do users ever see it? Chrome opens a new tab for me until the download is successful then I get the success message.

Reply:
They do but it might not be necessary. I think it was more useful when I was testing, I'll remove it.

Reply:
I see - you removed the functionality that opens the new window - Its not typical of CORe to show "in-progress" toast messages, but I guess it doesn't hurt.

Reply:
`downloadFileFromDocumentManager` still uses the "open new window" functionality